lua/nvim-treesitter/highlight.lua

```diff
@@ -114,13 +114,17 @@ function M.attach(bufnr, lang)
   local is_table = type(config.additional_vim_regex_highlighting) == "table"
   if config.additional_vim_regex_highlighting and (not is_table or config.additional_vim_regex_highlighting[lang]) then
     api.nvim_buf_set_option(bufnr, "syntax", "ON")
+  else
+    -- Prevent re-activating syntax when the file type is set/changed.
+    api.nvim_command(string.format("autocmd NvimTreesitter Syntax <buffer=%d> set syntax=", bufnr))
```

Comment:
doesn't this autocmd need a more unique name to not conflict with something else?

Reply:
Do you mean the `NvimTreesitter` name? That's the hl group that is defined to be used by this plugin

https://github.com/nvim-treesitter/nvim-treesitter/blob/2b785889a4d0613465f72eccabd42468b345658d/plugin/nvim-treesitter.vim#L7

If you mean the pattern, we can't be more specific because it can be changed to anything.

Reply:
Like `NvimTreesitterPreventSyntax` we only want to disable this autocmd on disable not something else. In the plugins we even included the exact purpose and buffer number for the autocmd.

Reply:
This will also prevent load syntax when the user explicitly requests that. We shouldn't resort to that hack that might be very unexpected for most users if there should should be another possibility to avoid this. Am I right that after this change `syntax` can only b brought back by disabling the Tree-Sitter highlight module?

Reply:
> Like NvimTreesitterPreventSyntax we only want to disable this autocmd on disable not something else. In the plugins we even included the exact purpose and buffer number for the autocmd.

Do you mean like this https://github.com/nvim-treesitter/playground/blob/520337f02d2b75bd8b7a997fac72ac889aeea89d/lua/nvim-treesitter-playground/query_linter.lua#L142-L162?

> Am I right that after this change syntax can only b brought back by disabling the Tree-Sitter highlight module?

Yes, I think that's expected as treesitter replaces vim's syntax, we already have an option to allow users to enable syntax.

Reply:
We have also more complicated names ;-). I don't know isn't the purpose of naming autocmds to refer to them later without having conflicts?  But I guess you know better how this works. Just double check whether it could have unwanted side effects like a different detach disabling this autocmds.